partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
 
-    validation_lab_cell_lines = (
-        biomarkers
-        # Selecting cell line name, cell line annotation and applyting the stacking expression:
-        .select(
-            f.col("cellName"),
-            "diseaseCellLines",
-            f.expr(biomarker_unstack),
+@dataclass
+class ValidationLabProjectParser:
+    diseaseFromSourceMappedId: str
+    diseaseFromSource: str
+    tissueId: str
+    experimentDataFile: str
+    biomarkerDataFile: str
+    studyOverview: str
+    primaryProjectId: str
+    releaseVersion: str
+    releaseDate: str
+    excludeStudy: bool
+
+    def parse_evidence(
+        self: ValidationLabProjectParser,
+        spark: SparkSession,
+        input_path: str,
+        assays: list,
+    ) -> DataFrame:
+        logger = logging.getLogger(__name__)
+
+        validation_input_file = f"{input_path}/{self.experimentDataFile}"
+        biomarker_file = f"{input_path}/{self.biomarkerDataFile}"
+
+        logger.info(f"Reading project data: {validation_input_file}")
+        logger.info(f"Reading biomarker data: {biomarker_file}")
+
+        # Reading data:
+        raw_experiment_data = spark.read.option("multiline", True).csv(
+            validation_input_file, sep="\t", header=True
         )
-        # Filter out all null biomarkers:
-        .filter(
-            (f.col("biomarkers").isNotNull())
-            &
-            # Following the request of the validation lab, we are removing CRIS biomarker annotation:
-            (f.col("biomarker_name") != "CRIS_subtype")
+        # First round of processing:
+        processed_experiment = self._process_raw_evidence(raw_experiment_data, assays)
+
+        # Reading biomarker data:
+        raw_biomarkers = spark.read.csv(biomarker_file, sep="\t", header=True)
+
+        # Formatting assays (depends on the formatted experiment data):
+        parsed_assays = self._parser_assay_object(processed_experiment, assays, spark)
+
+        # Get parsed biomarkers:
+        biomarkers = BiomarkerParser.get_biomarkers(raw_biomarkers)
+
+        # attributes added as column:
+        columns_to_add = [
+            "diseaseFromSourceMappedId",
+            "diseaseFromSource",
+            "tissueId",
+            "studyOverview",
+            "primaryProjectId",
+            "releaseVersion",
+            "releaseDate",
+        ]
+
+        return (
+            processed_experiment.join(
+                parsed_assays, on=["targetFromSourceId", "cellLineName"], how="inner"
+            )
+            .select(
+                # Columns from the raw evidence file:
+                "targetFromSourceId",
+                "cellLineName",
+                "assessment",
+                "resourceScore",
+                "primaryProjectHit",
+                # Column from the assay parser:
+                "assays",
+                # Columns from the project metadata:
```

Comment:
Aren't these already added later in the `generate_evidece` function?

Reply:
No, these are project specific constants not datasource specific ones. If I got your question right.